strings/check_palindrome.py

```diff
@@ -0,0 +1,57 @@
+# -*- coding: utf-8 -*-
+
+"""
+  check_palindrome.py: Checks if a string is a palindrome.
+
+  Author: Joaquin Cabezas github.com/joaquincabezas
+  Date: 07/04/2020
+
+"""
+
+def check_palindrome(input_string: str, exact_string: bool=True) -> bool:
+  """
+  A Palindrome String is a sequence of characters  which reads the same backward as forward.
+
+  Parameters
+  ----------
+  input_string : str
+      The string to be checked as a palindrome
+  exact_string : bool
+      If True: every character will be considered. 
+      If False: only letters will be considered
+        (thus removing spaces, punctuation and 
+        any other characters)
+  
+  Returns
+  -------
+  bool
+      True if the string is a palindrome
+
+  >>> check_palindrome("Rats live on no evil star")
+  True
+  >>> check_palindrome("Was it a car or a cat I saw?")
+  False
+  >>> check_palindrome("Was it a car or a cat I saw?", False)
+  True
+  >>> check_palindrome("A man, a plan, a canal, Panama.", False)
+  True
+  """
+
+  # There is no difference between lowercase and uppercase letters, so we convert everything to lowercase
+  input_string = input_string.lower()
+
+  # If we are not considering the full set of characters, we first remove everything except letters
+  if not exact_string:
+    # Only english characters
```

Comment:
You can `from string import ascii_lowercase` but perhaps https://docs.python.org/3/library/stdtypes.html#str.casefold would be even cooler.